// umi-plugin-ui/lib/plugins/dashboard/ui/plugins/dailyReport.js
"use strict";

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.MESSAGES = void 0;

var React = _interopRequireWildcard(require("react"));

var _swr = _interopRequireDefault(require("swr"));

var _classnames = _interopRequireDefault(require("classnames"));

var _antd = require("antd");

var _context3 = _interopRequireDefault(require("../context"));

var _dailyReportModule = _interopRequireDefault(require("./dailyReport.module.less"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { if (!(Symbol.iterator in Object(arr) || Object.prototype.toString.call(arr) === "[object Arguments]")) { return; } var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var Paragraph = _antd.Typography.Paragraph;
var useState = React.useState,
    useEffect = React.useEffect;
var PAGE_SIZE = 5;
var TAG_MAP = {
  发布: 'publish',
  文章: 'article',
  Star: 'star',
  star: 'star',
  Tweets: 'tweets'
};
var MESSAGES = {
  CHANGE_DAILY_ID: Symbol('CHANGE_DAILY_ID')
};
exports.MESSAGES = MESSAGES;

var DailyReport = function DailyReport(props) {
  var forceUpdate = props.forceUpdate;

  var _React$useContext = React.useContext(_context3.default),
      api = _React$useContext.api;

  var _ = api._,
      event = api.event;
  var formatMessage = api.intl.formatMessage;

  var _React$useState = React.useState(PAGE_SIZE),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      size = _React$useState2[0],
      setSize = _React$useState2[1];

  var _useSWR = (0, _swr.default)('zaobao.list', function _callee() {
    var _ref, data;

    return regeneratorRuntime.async(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _context.next = 2;
            return regeneratorRuntime.awrap(api.callRemote({
              type: 'org.umi.dashboard.zaobao.list'
            }));

          case 2:
            _ref = _context.sent;
            data = _ref.data;
            return _context.abrupt("return", data);

          case 5:
          case "end":
            return _context.stop();
        }
      }
    });
  }),
      list = _useSWR.data;

  var _useState = useState(),
      _useState2 = _slicedToArray(_useState, 2),
      currentId = _useState2[0],
      setCurrentId = _useState2[1];

  var changeCurrentId = function changeCurrentId(newId) {
    if (newId) {
      setCurrentId(newId);
      setSize(PAGE_SIZE);
    }
  };

  useEffect(function () {
    var id = _.get(list, '0.id');

    if (id) {
      setCurrentId(id);
    }
  }, [list]);
  useEffect(function () {
    event.on(MESSAGES.CHANGE_DAILY_ID, changeCurrentId);
    return function () {
      event.off(MESSAGES.CHANGE_DAILY_ID, changeCurrentId);
    };
  }, []);

  var _useSWR2 = (0, _swr.default)(function () {
    return "zaobao.list.detail.".concat(currentId);
  }, function _callee2(query) {
    var id, _ref2, _data;

    return regeneratorRuntime.async(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            id = Number(query.replace('zaobao.list.detail.', ''));

            if (!id) {
              _context2.next = 7;
              break;
            }

            _context2.next = 4;
            return regeneratorRuntime.awrap(api.callRemote({
              type: 'org.umi.dashboard.zaobao.list.detail',
              payload: {
                id: id
              }
            }));

          case 4:
            _ref2 = _context2.sent;
            _data = _ref2.data;
            return _context2.abrupt("return", _data);

          case 7:
          case "end":
            return _context2.stop();
        }
      }
    });
  }),
      data = _useSWR2.data;

  useEffect(function () {
    forceUpdate();
  }, [data]);
  var length = Array.isArray(data) ? data.length : 0;

  var handleLoadMore = function handleLoadMore() {
    setSize(function (value) {
      return value + PAGE_SIZE;
    }); // 重新计算瀑布流

    forceUpdate();
  };

  var LoadMore = size < length && React.createElement("div", {
    style: {
      textAlign: 'center',
      marginTop: 12,
      height: 32,
      lineHeight: '32px'
    }
  }, React.createElement("a", {
    className: _dailyReportModule.default.more,
    onClick: handleLoadMore
  }, formatMessage({
    id: 'org.umi.ui.dashboard.card.zaobao.loadMore'
  })));

  var getTagCls = function getTagCls(name) {
    return (0, _classnames.default)(_dailyReportModule.default['listItem-meta-tag'], _defineProperty({}, _dailyReportModule.default["listItem-meta-tag-".concat(TAG_MAP[name] || TAG_MAP.Star)], !!name));
  };

  return React.createElement(_antd.List, {
    itemLayout: "horizontal",
    loading: !data,
    className: _dailyReportModule.default.list,
    split: false,
    dataSource: Array.isArray(data) ? data.slice(0, size) : data,
    loadMore: LoadMore,
    renderItem: function renderItem(item) {
      return _.isPlainObject(item) && React.createElement(_antd.List.Item, {
        className: _dailyReportModule.default.listItem
      }, React.createElement(_antd.List.Item.Meta, {
        className: _dailyReportModule.default['listItem-meta'],
        title: React.createElement("a", {
          target: "_blank",
          rel: "noopener noreferrer",
          href: item.href
        }, React.createElement(Paragraph, {
          className: _dailyReportModule.default.paragraph,
          ellipsis: true
        }, item.title), ' ', React.createElement(_antd.Tag, {
          className: getTagCls(item.tag)
        }, item.tag)),
        description: item.description
      }));
    }
  });
};

var _default = DailyReport;
exports.default = _default;
