// umi-plugin-ui/lib/plugins/dashboard/ui/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.MESSAGES = exports.renderAvatar = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactMasonryComponent = _interopRequireDefault(require("react-masonry-component"));

var _isPlainObject = _interopRequireDefault(require("lodash/isPlainObject"));

var _antd = require("antd");

var _context = _interopRequireDefault(require("./context"));

var _indexModule = _interopRequireDefault(require("./index.module.less"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { if (!(Symbol.iterator in Object(arr) || Object.prototype.toString.call(arr) === "[object Arguments]")) { return; } var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useCallback = _react.default.useCallback,
    useEffect = _react.default.useEffect;

var renderAvatar = function renderAvatar(item) {
  var isMini = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
  var icon = item.icon,
      title = item.title,
      _item$backgroundColor = item.backgroundColor,
      backgroundColor = _item$backgroundColor === void 0 ? '#459BF7' : _item$backgroundColor;
  var commonAvatarProps = {
    style: {
      backgroundColor: backgroundColor
    },
    shape: 'square',
    className: _indexModule.default.avatar,
    size: isMini ? 'small' : 'default'
  };

  if (_react.default.isValidElement(icon)) {
    return _react.default.createElement(_antd.Avatar, _extends({}, commonAvatarProps, {
      icon: icon
    }));
  }

  if (typeof icon === 'string') {
    return _react.default.createElement(_antd.Avatar, _extends({}, commonAvatarProps, {
      src: icon
    }));
  }

  if ((0, _isPlainObject.default)(icon)) {
    return _react.default.createElement(_antd.Avatar, _extends({}, commonAvatarProps, icon));
  } // 默认返回 title 第一个字符


  var defaultChar = typeof title === 'string' ? title.charAt(0) : 'Umi';

  if (defaultChar) {
    return _react.default.createElement(_antd.Avatar, commonAvatarProps, defaultChar);
  }
};

exports.renderAvatar = renderAvatar;
var MESSAGES = {
  CHANGE_CARDS: Symbol('CHANGE_CARDS')
};
exports.MESSAGES = MESSAGES;
var DEFAULT_SPAN = {
  xs: 24,
  sm: 12,
  lg: 12,
  xl: 6
};
var DEFAULT_SPAN_MINI = {
  xs: 12,
  sm: 8,
  lg: 8,
  xl: 8
};

var DashboardUI = function DashboardUI(props) {
  // const isClosed = window.localStorage.getItem('umi_ui_dashboard_welcome') || false;
  // const [closed, setClosed] = useState<boolean>(!!isClosed);
  var _React$useContext = _react.default.useContext(_context.default),
      api = _React$useContext.api,
      loading = _React$useContext.loading,
      dashboardCards = _React$useContext.dashboardCards,
      setCardSettings = _React$useContext.setCardSettings;

  var event = api.event,
      currentProject = api.currentProject,
      intl = api.intl,
      _api$getBasicUI = api.getBasicUI,
      getBasicUI = _api$getBasicUI === void 0 ? function () {
    return {};
  } : _api$getBasicUI;

  var _React$useState = _react.default.useState(),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      updateState = _React$useState2[1];

  var forceUpdate = useCallback(function () {
    return updateState({});
  }, []);
  var actionCardCls = (0, _classnames.default)(_indexModule.default.card, _indexModule.default['card-action']);
  var welcomeCardCls = (0, _classnames.default)(_indexModule.default.card, _indexModule.default.welcome);
  var basicUI = getBasicUI();
  var containerCls = (0, _classnames.default)(_indexModule.default['container-row'], 'ant-row', 'ant-row-flex');

  var renderCard = function renderCard(content) {
    var childProps = {
      // 用于动态卡片 DidMount 后，重新计算瀑布流
      forceUpdate: forceUpdate
    };

    if (Array.isArray(content)) {
      // 横向排列快捷入口
      return _react.default.createElement(_antd.Row, {
        className: _indexModule.default['content-row'],
        type: "flex"
      }, _react.default.Children.map(content, function (child, i) {
        return _react.default.createElement(_antd.Col, {
          className: _indexModule.default['content-col'],
          key: i.toString(),
          span: Math.floor(24 / content.length)
        }, child);
      }));
    }

    if (_react.default.isValidElement(content)) {
      return _react.default.cloneElement(content, childProps);
    }
  };

  useEffect(function () {
    event.on(MESSAGES.CHANGE_CARDS, setCardSettings);
    return function () {
      event.off(MESSAGES.CHANGE_CARDS, setCardSettings);
    };
  }, []);
  var enableCards = (dashboardCards || []).filter(function (card) {
    return !!card.enable;
  });

  if (loading) {
    return _react.default.createElement(_antd.Spin, null);
  }

  var defaultSpan = api.mini ? DEFAULT_SPAN_MINI : DEFAULT_SPAN;
  var colCls = (0, _classnames.default)(actionCardCls, _indexModule.default['container-col']);
  return _react.default.createElement("div", {
    className: _indexModule.default.container
  }, _react.default.createElement(_reactMasonryComponent.default, {
    className: containerCls
  }, _react.default.createElement(_antd.Col, _extends({
    className: colCls
  }, defaultSpan), _react.default.createElement(_antd.Card, {
    className: welcomeCardCls,
    bordered: false,
    hoverable: false
  }, _react.default.createElement("h2", null, "Hi"), _react.default.createElement("p", null, intl({
    id: 'org.umi.ui.dashboard.panel.welcome.title'
  }, {
    name: currentProject.name
  })), _react.default.createElement("div", null, _react.default.createElement("a", {
    href: "https://umijs.org/guide/umi-ui.html",
    target: "_blank",
    rel: "noopener noreferrer"
  }, basicUI.name || 'Umi', " UI"), intl({
    id: 'org.umi.ui.dashboard.panel.welcome.desc'
  })))), enableCards.map(function (card) {
    var _card$title = card.title,
        title = _card$title === void 0 ? '' : _card$title,
        description = card.description,
        key = card.key,
        content = card.content,
        _card$span = card.span,
        span = _card$span === void 0 ? defaultSpan : _card$span,
        right = card.right;

    var Title = _react.default.createElement(_antd.Row, {
      className: _indexModule.default.main,
      type: "flex",
      gutter: 16,
      align: "top",
      justify: "space-between"
    }, _react.default.createElement(_antd.Col, {
      className: _indexModule.default['main-col']
    }, _react.default.createElement("div", {
      className: (0, _classnames.default)(_indexModule.default.icon)
    }, renderAvatar(card, api.mini)), _react.default.createElement("div", {
      className: _indexModule.default.info
    }, _react.default.createElement("h4", null, title), description && _react.default.createElement("p", null, description))), right && _react.default.createElement("div", {
      className: _indexModule.default.right
    }, right));

    var colSpan = api._.isPlainObject(span) ? _objectSpread({}, defaultSpan, {}, span) : {
      span: span
    };
    return _react.default.createElement(_antd.Col, _extends({
      key: key,
      className: colCls
    }, colSpan), _react.default.createElement(_antd.Card, {
      title: Title,
      className: _indexModule.default.card,
      bordered: false,
      hoverable: false
    }, renderCard(content)));
  })));
};

var _default = DashboardUI;
exports.default = _default;
