// umi-plugin-ui/lib/plugins/dashboard/ui/setting.js
"use strict";

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _antd = require("antd");

var _context2 = _interopRequireDefault(require("./context"));

var _index = require("./index");

var _settingModule = _interopRequireDefault(require("./setting.module.less"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Paragraph = _antd.Typography.Paragraph;

var Setting = function Setting(props, ref) {
  var _React$useContext = React.useContext(_context2.default),
      api = _React$useContext.api,
      dbPath = _React$useContext.dbPath,
      cards = _React$useContext.cards,
      setCardSettings = _React$useContext.setCardSettings;

  var _ref$current = ref.current,
      visible = _ref$current.visible,
      closeDrawer = _ref$current.closeDrawer,
      openDrawer = _ref$current.openDrawer,
      className = _ref$current.className,
      restConfig = _objectWithoutProperties(_ref$current, ["visible", "closeDrawer", "openDrawer", "className"]);

  var handleOnChange = function handleOnChange(key, checked) {
    var result;
    return regeneratorRuntime.async(function handleOnChange$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _context.next = 2;
            return regeneratorRuntime.awrap(api.callRemote({
              type: 'org.umi.dashboard.card.list.change',
              payload: {
                dbPath: dbPath,
                key: key,
                enable: !!checked
              }
            }));

          case 2:
            result = _context.sent;
            api.event.emit(_index.MESSAGES.CHANGE_CARDS, result);
            setCardSettings(result);

          case 5:
          case "end":
            return _context.stop();
        }
      }
    });
  };

  var drawerCls = (0, _classnames.default)(className, _settingModule.default.drawer);
  return React.createElement(_antd.Drawer, _extends({
    className: drawerCls,
    visible: visible,
    onClose: closeDrawer,
    title: api.intl({
      id: 'org.umi.ui.dashboard.settings.title'
    })
  }, restConfig), React.createElement(_antd.List, {
    dataSource: cards || [],
    loading: !cards,
    className: _settingModule.default.list,
    bordered: false,
    split: false,
    renderItem: function renderItem(item) {
      return React.createElement(_antd.List.Item, {
        key: item.key,
        className: _settingModule.default.item,
        extra: React.createElement(_antd.Switch, {
          style: {
            marginLeft: 8
          },
          size: "small",
          onChange: function onChange(checked) {
            return handleOnChange(item.key, checked);
          },
          defaultChecked: !!item.enable
        })
      }, React.createElement(_antd.List.Item.Meta, {
        avatar: (0, _index.renderAvatar)(item, api.mini),
        title: item.title,
        description: React.createElement(Paragraph, {
          ellipsis: true
        }, item.description)
      }));
    }
  }));
};

var _default = React.forwardRef(Setting);

exports.default = _default;
